ci/ansible/playbooks/os/redhat/upgrade_release.yml

```diff
@@ -0,0 +1,378 @@
+      command: 'getenforce'
+      register: pre_upgrade_selinux_mode
+
+    - name: Register active repositories
+      command: "yum repolist"
+      register: pre_upgrade_active_repositories
+
+    - name: Ensure repositories that provide leapp utility are enabled
+      ini_file:
+        path: "{{ item.repo_file }}"
+        section: "{{ item.name }}"
+        option: enabled
+        value: 1
+      loop: "{{ leapp_dependencies.repos[provider] }}"
+
+    - &UPDATE_PACKAGES
+      name: Update all packages in current major version
+      yum:
+        update_cache: true
+        name: "*"
+        state: latest
+
+    - name: Reboot system # to load kernel from latest minor version if any
+      reboot:
+        msg: Reboot initiated by Ansible due to release upgrade
+        connect_timeout: "{{ reboot.connect_timeout }}"
+        reboot_timeout: "{{ reboot.update_timeout }}"
+
+    - name: Assert latest minor version
+      assert:
+        that: ansible_distribution_version == "{{ required_minor_version }}"
+        quiet: true
+
+    - name: Install packages that provide the leapp utility
+      package:
+        name: "{{ leapp_dependencies.packages[provider] }}"
+        state: present
+
+    # The following leapp_metadata archive is tied to a subscription,
+    # thus it has been encrypted to avoid licensing conflicts
+    - name: Create leapp metadata archive
+      copy:
+        content: "{{ metadata_archive | b64decode }}"
+        dest: /etc/leapp/files/leapp-data15.tar.gz
+
+    - name: Unarchive leapp metadata
+      unarchive:
+        src: /etc/leapp/files/leapp-data15.tar.gz
+        dest: /etc/leapp/files/
+        remote_src: true
+
+    # Address upgrade inhibitors
+    - name: Disable ssh root login
+      lineinfile:
+        dest: /etc/ssh/sshd_config
+        regexp: '^PermitRootLogin'
+        line: 'PermitRootLogin no'
+        state: present
+      register: sshd_config
+
+    - &RESTART_SSHD
+      name: Restart ssh server
+      when: sshd_config.changed
+      service:
+        name: sshd
+        state: restarted
+
+    - name: Disable kernel modules that are not supported in RHEL8
+      modprobe:
+        name: "{{ item }}"
+        state: absent
+      loop:
+        - pata_acpi
+        - floppy
+
+    - name: Keep only one kernel-devel package when multiple are present
+      block:
+        - name: List duplicated kernel-devel packages
+          yum:
+            list: kernel-devel
+          register: kernel_devel_packages
+
+        - name: Register version of installed packages
+          set_fact:
+            installed_kernel_devel_packages: "{{ kernel_devel_packages|json_query(query) }}"
+          vars:
+            query: "results[?yumstate=='installed'].envra"
+
+        - name: Remove only duplicates
+          when: installed_kernel_devel_packages | length > 1
+          yum:
+            name: "{{ installed_kernel_devel_packages[1:] }}"
+            state: absent
+
+    ### UPGRADE ###
```

Comment:
This playbook is meant to be run in CI so I would exceptionally vote for limiting amount of files since they will be downloaded from GH using curl to avoid cloning whole repo.